File: Day3/Part1.py
total = 0

with open("C:\\Users\\User\\Downloads\\advent.txt") as file:
    for line in file:
        items = line.strip()

        first = items[:len(items)//2]
        second = items[len(items)//2:]

        value = 0
        for item in first:
            if item in second:
                value = ord(item)
                break

        if value > 97:
            value -= 96
        else:
            value -= 38

        total += value

print(total)

File: src/2023/src/Day01/Part1.java
package Day1;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class Part1 {

    public static void main(String[] args) throws IOException {
        String file = "C:\\Users\\User\\Downloads\\advent.txt";

        int total = 0;
        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
            for(String line; (line = reader.readLine()) != null; ) {
                line = line.replaceAll("[^0-9]+", "");

                total += Integer.parseInt("" + line.charAt(0) + line.charAt(line.length() - 1));
            }
        }

        System.out.println(total);
    }
}
